### ISSUE TITLE
systemctl --user exit requires dbus

### ISSUE THREAD

**Issue by keszybz** [systemd/systemd#10414]
https://github.com/systemd/systemd/pull/8648 changed `kill -s 58 $MANAGERPID` to `systemctl --user --force exit`, but this requires dbus to be present. In particular, it it possible for the user's `dbus.service` to be socket activated through `dbus.socket`, which means that we would trigger a job to start a unit when stopping. It seems this change should be reverted and we should just send the signal directly from `systemctl`.

**Comment by keszybz** [comment]
```
Oct 16 07:07:30 fcloud2 systemd[6898]: Closed D-Bus User Message Bus Socket.
Oct 16 07:07:30 fcloud2 systemd[6898]: Reached target Shutdown.
Oct 16 07:07:30 fcloud2 systemd[6898]: Starting Exit the Session...
Oct 16 07:07:30 fcloud2 systemctl[7076]: Failed to connect to bus: Connection refused
Oct 16 07:07:30 fcloud2 systemd[6898]: systemd-exit.service: Main process exited, code=exited, status=1/FAILURE
Oct 16 07:07:30 fcloud2 systemd[6898]: systemd-exit.service: Failed with result 'exit-code'.
Oct 16 07:07:30 fcloud2 systemd[6898]: Failed to start Exit the Session.
Oct 16 07:07:30 fcloud2 systemd[6898]: Dependency failed for Exit the Session.
Oct 16 07:07:30 fcloud2 systemd[6898]: exit.target: Job exit.target/start failed with result 'dependency'.
```

**Comment by boucman** [comment]
is SuccessAction=exit a thing ? it doesn't appear to be documented afaict... (just a quick confirmation and i'll open an issue about that)

**Comment by keszybz** [comment]
I like this. I'll whip up a patch.

**Comment by poettering** [comment]
oh, and not just systemd-exit.service should be updated with this then, but systemd-reboot.service and so on, too

**Comment by poettering** [comment]
> is SuccessAction=exit a thing ? it doesn't appear to be documented afaict... (just a quick confirmation and i'll open an issue about that)

https://www.freedesktop.org/software/systemd/man/systemd.unit.html#FailureAction=

(Looking at that man page I think the explanatory paragraph really could need some love though)

**Comment by jnpkrn** [comment]
Related to the fix for #9461, I am now observing this problem with v239:

- in a virtual terminal as a user, log in, log out

- reboot/poweroff the machine by non-command-line means
  (C-A-D/power off button)

- the termination process stuck on user manager for 2 minutes
  (`TimeoutStopSec`)

- meanwhile in the logs:

```
09:56:24 systemd[1]: Stopping User Manager for UID 1000...
09:56:24 systemd[2139]: Received SIGTERM from PID 1 (systemd).
[...]
09:56:24 systemctl[3957]: systemd-exit.service: Executing: /usr/bin/systemctl --user --force exit
09:56:24 systemctl[3957]: Failed to connect to bus: No such file or directory
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
09:56:24 systemd[2139]: Received SIGCHLD from PID 3957 (systemctl).
09:56:24 systemd[2139]: Child 3957 (systemctl) died (code=exited, status=1/FAILURE)
09:56:24 systemd[2139]: systemd-exit.service: Child 3957 belongs to systemd-exit.service.
09:56:24 systemd[2139]: systemd-exit.service: Main process exited, code=exited, status=1/FAILURE
09:56:24 systemd[2139]: systemd-exit.service: Failed with result 'exit-code'.
09:56:24 systemd[2139]: systemd-exit.service: Changed start -> failed
09:56:24 systemd[2139]: systemd-exit.service: Job systemd-exit.service/start finished, result=failed
09:56:24 systemd[2139]: exit.target: Job exit.target/start finished, result=dependency
09:56:24 systemd[2139]: exit.target: Job exit.target/start failed with result 'dependency'.
09:56:24 systemd[2139]: systemd-exit.service: Unit entered failed state.
09:56:24 systemd[2139]: exit.target: Collecting.
09:58:24 systemd[1]: user@1000.service: State 'stop-sigterm' timed out. Killing.
09:58:24 systemd[1]: user@1000.service: Killing process 2139 (systemd) with signal SIGKILL.
09:58:24 systemd[1]: user@1000.service: Failed with result 'timeout'.
09:58:24 systemd[1]: Stopped User Manager for UID 1000.
```

Since `exit-force` action is not available on that version, I resorted
to previous solution per add384dd4d2b96db6ace5ad9c52b1dd7553ebec2 that
doesn't appear entirely graceful but doesn't incur said blocking.

Looking forward to this fix.

**Comment by keszybz** [comment]
Hmm, we should either revert add384d in v239-stable or backport those patches.
